src/main.c

```diff
@@ -733,6 +734,30 @@ static m64p_error ParseCommandLineMain(int argc, const char **argv)
             ConfigSetParameter(l_Config64DD, "Disk", M64TYPE_STRING, argv[i+1]);
             i++;
         }
+        else if (strcmp(argv[i], "--pif") == 0)
+        {
+            m64p_error pif_status = M64ERR_INVALID_STATE;
+            /* load PIF image */
+            FILE *pifPtr = fopen(argv[i+1], "rb");
+            if (pifPtr != NULL)
+            {
+                unsigned char *PIF_buffer = (unsigned char *) malloc(2048);
+                if (PIF_buffer != NULL)
+                {
+                    if (fread(PIF_buffer, 1, 2048, pifPtr) == 2048)
+                    {
+                        pif_status = (*CoreDoCommand)(M64CMD_PIF_OPEN, 2048, PIF_buffer);
+                    }
+                    free(PIF_buffer);
+                }
+                fclose(pifPtr);
```

Comment:
Not a objection, mostly a discussion: Use a constant (`PIF_ROM_SIZE`?) instead of a  magic number?

Then, as it's a file wouldn't it been better to allocate the file size? [Like we do for the ROM cart?](https://github.com/mupen64plus/mupen64plus-ui-console/blob/1c35ee27467ccd1b6f219b0dda0e9c94b49c52e3/src/main.c#L1001) Just in case we would have custom PIF ROMs (from hardware hackers).

Reply:
Yea a constant would make sense. It's a bit different from a cartridge ROM though. Cartridges have variable lengths, the biggest commerical ROMs were 64MB, but it seems like the theoretical limit would be 264MB, based on how the memory addressing is laid out.

Because of the N64 address layout, the PIF ROM cannot be larger than 2048 bytes, so we can be pretty confident that even a customized version would be this size.